### ISSUE TITLE
Intl doesn't work on iOS [RN 76 + Expo 52 bare]

### ISSUE THREAD

**Issue by FelipeSSantos1** [facebook/react-native#47640]
### Description

`Intl.DateTimeFormat()` is returning an empty object on iOS, it only happens on RN 76.2

### Steps to reproduce

```
const date = new Date();
  const timeZoneOptions = Intl.DateTimeFormat().resolvedOptions();
  const timezone = timeZoneOptions.timeZone;
  const shortFormatter = new Intl.DateTimeFormat("en-US", {
    timeZone: timezone,
    timeZoneName: "short",
  });

  const shortParts = shortFormatter.formatToParts(date);
```
error: shortFormatter.formatToParts is not a function (it is undefined)

### React Native Version

0.76.1

### Affected Platforms

Runtime - iOS

### Output of `npx react-native info`

```text
System:
  OS: macOS 14.7
  CPU: (10) arm64 Apple M1 Pro
  Memory: 806.14 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 22.9.0
    path: ~/.nvm/versions/node/v22.9.0/bin/node
  Yarn:
    version: 1.22.22
    path: /opt/homebrew/bin/yarn
  npm:
    version: 10.8.3
    path: ~/.nvm/versions/node/v22.9.0/bin/npm
  Watchman:
    version: 2024.10.14.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /Users/felipe.santos/.rvm/gems/ruby-3.2.4/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.5
      - iOS 17.5
      - macOS 14.5
      - tvOS 17.5
      - visionOS 1.2
      - watchOS 10.5
  Android SDK:
    API Levels:
      - "28"
      - "30"
      - "31"
      - "33"
      - "34"
      - "35"
    Build Tools:
      - 30.0.2
      - 30.0.3
      - 31.0.0
      - 33.0.0
      - 33.0.1
      - 33.0.2
      - 34.0.0
      - 35.0.0
    System Images:
      - android-30 | ARM 64 v8a
      - android-30 | Google APIs ARM 64 v8a
      - android-30 | Google Play ARM 64 v8a
      - android-30 | Google APIs ATD ARM 64 v8a
      - android-31 | ARM 64 v8a
      - android-31 | Google TV ARM 64 v8a
      - android-31 | Google APIs ARM 64 v8a
      - android-31 | Google Play ARM 64 v8a
      - android-33 | Intel x86_64 Atom
      - android-33 | Google APIs ARM 64 v8a
      - android-33 | Google Play ARM 64 v8a
      - android-34 | Intel x86_64 Atom
      - android-34 | Google APIs ARM 64 v8a
      - android-34 | Google Play ARM 64 v8a
      - android-35 | Google Play ARM 64 v8a
    Android NDK: Not Found
IDEs:
  Android Studio: 2024.2 AI-242.23339.11.2421.12550806
  Xcode:
    version: 15.4/15F31d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.13
    path: /usr/bin/javac
  Ruby:
    version: 3.2.4
    path: /Users/felipe.santos/.rvm/rubies/ruby-3.2.4/bin/ruby
npmPackages:
  "@react-native-community/cli":
    installed: 15.0.1
    wanted: 15.0.1
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.76.2
    wanted: 0.76.2
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false
```
```


### Stacktrace or Logs

```text
shortFormatter.formatToParts is not a function (it is undefined)
```

### Reproducer

https://snack.expo.dev/DapJeitRI6jHU6ay4lCMe?platform=ios

**Comment by FelipeSSantos1** [comment]
it is fixed on RN `0.76.2`

**Comment by maksim-romanov** [comment]
a same issue for me, i am using 0.76.5, `formatRange` is `undefined`
